lib/mix/test/mix/tasks/compile.app_test.exs

```diff
@@ -3,6 +3,12 @@ Code.require_file("../../test_helper.exs", __DIR__)
 defmodule Mix.Tasks.Compile.AppTest do
   use MixTest.Case
 
+  def parse_resource_file(app) do
+    {:ok, [term]} = :file.consult("_build/dev/lib/#{app}/ebin/#{app}.app")
+    {:application, ^app, properties} = term
+    properties
+  end
```

Comment:
Personally I prefer to put helper functions at the bottom of the file, after all tests. But regardless of personal preference, this should be private.